sdks/python/apache_beam/runners/portability/fn_api_runner/fn_runner.py

```diff
@@ -511,87 +509,32 @@ def _add_residuals_and_channel_splits_to_deferred_inputs(
 
   def _run_stage(self,
                  runner_execution_context,  # type: execution.FnApiRunnerExecutionContext
-                 stage,  # type: translations.Stage
+                 bundle_context_manager,  # type: execution.BundleContextManager
```

Comment:
It's odd that _run_stage no longer takes as a parameter the stage to run. Perhaps bundle_context_manager (and its class?) should be named stage_context or similar?

Reply:
On this note, perhaps it makes sense to break FnApiRunner into the (mostly stateless) runner that can execute multiple pipelines and an executor (that has methods like run_stage) that might be stateful and is initialized with and tasked with running a single pipeline. Much of what is on context(s) would become state of self of this new object.

Reply:
I'm adding all these comments to https://github.com/apache/beam/pull/11270